src/eval.rs

```diff
@@ -703,10 +721,10 @@ fn reduce_with_witness_inner<F: LurkField>(
                             }
                         }
                     } else if head == store.lurk_sym("cons") {
-                        let (arg1, more) =
-                            cons_witness.car_cdr_named(ConsName::ExprCdr, store, &rest)?;
-                        if more.is_nil() {
-                            Control::Error(arg1, env)
+                        let (arg1, more) = car_cdr_named!(ConsName::ExprCdr, &rest)?;
+
+                        if rest.is_nil() || more.is_nil() {
```

Comment:
I'm a bit confused: if `rest.is_nil()`, doesn't `car_cdr_named` return `Err(ReductionError::CarCdrType(name)` anyway?

Reply:
No, `(car nil)` and `(cdr nil)` both return `NIL`. That's why I didn't named the error `BadCons` (since NIL isn't a cons).